docs/fundamentals/code-analysis/style-rules/language-rules.md

```diff
@@ -0,0 +1,1920 @@
+---
+title: Code style language rules
+ms.date: 09/25/2020
+ms.topic: reference
+author: gewarren
+ms.author: gewarren
+dev_langs:
+- CSharp
+- VB
+helpviewer_keywords:
+- language code style rules [EditorConfig]
+- language rules
+- EditorConfig language conventions
+---
+# Language rules
+
+Code style language rules affect how various constructs of .NET programming languages are used, for example, modifiers and parentheses. They fall into the below categories:
+
+- [.NET style rules](#net-style-rules): Rules that apply to both C# and Visual Basic. The EditorConfig option names for these rules start with `dotnet_style_` prefix.
+- [C# style rules](#c-style-rules): Rules that are specific to C# language only. The EditorConfig option names for these rules start with `csharp_style_` prefix.
+
+> [!TIP]
+>
+> - To see the code examples in your preferred programming language, choose it using the language picker at the top-right corner of the browser window.
+>
+>   ![Code language picker control](media/code-language-picker.png)
+>
+> - Use the **In this article** links to jump to different sections of the page.
+
+## Option format
+
+Options for language rules can be specified in an EditorConfig file with the following format:
+
+`option_name = value:severity`
```

Comment:
I think it will be confusing to show this as the syntax, but then to say "but it won't work if you're not using VS". I think that would be fine in the VS docs, but now that this is in the .NET docs, we should make the acceptable syntax more visible. Customers aren't necessarily going to read the paragraphs; they're going to jump to the code blocks and copy them. As one of the commenters in the Roslyn issue you linked to said, it now requires two lines of configuration for the compilers to respect the rule, correct? One to say if the rule is preferred, and one to set the severity? If that's the case, I think we should include both of those lines here. Then in the paragraph, we can say something like "if you're using VS, the second line isn't necessary".

Reply:
Sure, I can add the second line. Its really unfortunate this is so messed up that any documentation here is going to be confusing, highlighting that the design itself is bad. I am going to try to push forward on https://github.com/dotnet/roslyn/issues/44201 so we can just delete mentioning/requiring the severity part in this entry.

Reply:
I am still thinking about this, and I think adding a second line here for `dotnet_diagnostic.ruleId.severity = severity` is actually going to add more confusion. This page is about configuring option values, and making them write 2 entries with severity values seems confusing. I don't think there is a good way to document this properly unless `option_name = value` is made a legal syntax, at which point we can leave out all severity related stuff from this page. I am wondering if it is fine to leave this here for now, and then remove it in a follow-up PR once I make the Roslyn change to support it. Does that sound reasonable @gewarren?

Reply:
Yes, that sounds reasonable, thanks.